test/e2e/field_selectors/field_selectors_test.go

```diff
@@ -0,0 +1,181 @@
+	"strings"
+	"testing"
+
+	"github.com/flightctl/flightctl/test/harness/e2e"
+	"github.com/flightctl/flightctl/test/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+const TIMEOUT = "1m"
+const POLLING = "250ms"
+
+func TestFieldSelectors(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Field Selectors E2E Suite")
+}
+
+var _ = Describe("Field Selectors in Flight Control", func() {
+
+	var (
+		harness *e2e.Harness
+		token   string
+	)
+
+	BeforeEach(func() {
+		harness = e2e.NewTestHarness()
+
+		// construct the flightctl login arguments
+		loginArgs := []string{"login", "${API_ENDPOINT}", "--insecure-skip-tls-verify"}
+		if token != "" {
+			loginArgs = append(loginArgs, "--token", token)
+		}
+		// attempt login
+		out, err := harness.CLI(loginArgs...)
+
+		// if openshift authentication is required, try to obtain a token
+		if strings.Contains(out, "You must obtain an API token by visiting") {
+			token, err = harness.SH("oc", "whoami", "-t")
+			token = strings.Trim(token, "\n")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(token).ToNot(BeEmpty(), "token from oc whoami should not be empty")
+			loginArgs = append(loginArgs, "--token", token)
+			out, err = harness.CLI(loginArgs...)
+
+		}
+		Expect(err).ToNot(HaveOccurred())
+		out = strings.Trim(out, "\n.")
+		Expect(out).To(BeElementOf("Auth is disabled", "Login successful"))
+	})
+
+	AfterEach(func() {
+		harness.Cleanup(false) // do not print console on error
+	})
+
+	Context("login", func() {
+		It("should have worked, and we can list devices", func() {
+			out, err := harness.CLI("get", "devices")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("NAME"))
+		})
+	})
+
+	Context("apply/recursive", func() {
+		It("should work for a complete set of yamls", func() {
+			out, err := harness.CLI("apply", "-R", "-f", filepath.Join(util.GetTopLevelDir(), "examples/device.yaml"))
+			Expect(err).ToNot(HaveOccurred())
+			// expect out to contain 200 OK or 201 Created
+			Expect(out).To(MatchRegexp(`(200 OK|201 Created)`))
+
+			// check a for a couple of the yamls we know to exist in the examples directory
+			Expect(out).To(ContainSubstring("examples/device.yaml/f62dfb4f5d2cdbb9339362b5f18f2ce268d75649bdc80003f0e04292a7ef8c3"))
+			Expect(out).To(ContainSubstring("examples/resourcesync.yaml/default-sync"))
+		})
+	})
+
+	Context("apply/fleet", func() {
+		It("create for a complete fleet", func() {
+			// make sure it doesn't exist
+			_, _ = harness.CLI("delete", "fleet/e2e-test-fleet")
+
+			out, err := harness.CLI("apply", "-f", filepath.Join(util.GetTopLevelDir(), "examples/fleet.yaml"))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(out).To(ContainSubstring("201 Created"))
+		})
+	})
+
+	Context("Basic Functionality Tests", func() {
+		It("filters devices by name", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "metadata.name=f62dfb4f5d2cdbb9339362b5f18f2ce268d75649bdc80003f0e04292a7ef8c3")
```

Comment:
Could you, please use variable for the device name (here and other places in this PR). It will improve the readability and reduce the effort if the editing id needed.